runtime/util/hshelp.c

```diff
@@ -932,12 +932,14 @@ static UDATA
 findMethodInVTable(J9Method *method, UDATA *vTable)
 {
 	J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
-	UDATA vTableSize = *vTable;
-	UDATA searchIndex;
+	J9VTableHeader *vTableHeader = (J9VTableHeader *)vTable;
+	UDATA vTableSize = vTableHeader->size;
+	UDATA searchIndex = 0;
+	UDATA *vTableMethods = J9VTABLE_FROM_HEADER(vTableHeader);
 
 	/* Search the vTable for a public method of the correct name. */
-	for (searchIndex = 2; searchIndex <= vTableSize; searchIndex++) {
-		J9Method *vTableMethod = (J9Method *)vTable[searchIndex];
+	for (searchIndex; searchIndex < vTableSize; searchIndex++) {
```

Comment:
searchIndex; does nothing. It would be much clearer to initialize it to 0 again here.